src/editor/mod.rs

```diff
@@ -37,52 +37,75 @@ pub struct Editor {
+                Some(colors::WHITE),
+                Some(colors::BLACK),
+                Some(colors::GREY),
+            ))),
             defined_styles: HashMap::new(),
-            previous_style: None
+            previous_style: None,
         };
 
         editor.grid.clear();
         editor
     }
 
-
     pub fn handle_redraw_event(&mut self, event: RedrawEvent) {
         match event {
             RedrawEvent::SetTitle { title } => self.title = title,
             RedrawEvent::ModeInfoSet { cursor_modes } => self.cursor.mode_list = cursor_modes,
             RedrawEvent::OptionSet { gui_option } => self.set_option(gui_option),
-            RedrawEvent::ModeChange { mode_index } => self.cursor.change_mode(mode_index, &self.defined_styles),
+            RedrawEvent::ModeChange { mode_index } => {
+                self.cursor.change_mode(mode_index, &self.defined_styles)
+            }
             RedrawEvent::BusyStart => {
                 trace!("Cursor off");
                 self.cursor.enabled = false;
-            },
+            }
             RedrawEvent::BusyStop => {
                 trace!("Cursor on");
                 self.cursor.enabled = true;
-            },
+            }
             RedrawEvent::Flush => {
                 trace!("Image flushed");
                 REDRAW_SCHEDULER.queue_next_frame();
-            },
+            }
             RedrawEvent::Resize { width, height, .. } => self.grid.resize(width, height),
-            RedrawEvent::DefaultColorsSet { colors } => self.default_style = Arc::new(Style::new(colors)),
-            RedrawEvent::HighlightAttributesDefine { id, style } => { self.defined_styles.insert(id, Arc::new(style)); },
-            RedrawEvent::GridLine { row, column_start, cells, .. } => self.draw_grid_line(row, column_start, cells),
+            RedrawEvent::DefaultColorsSet { colors } => {
+                self.default_style = Arc::new(Style::new(colors))
+            }
+            RedrawEvent::HighlightAttributesDefine { id, style } => {
+                self.defined_styles.insert(id, Arc::new(style));
+            }
+            RedrawEvent::GridLine {
+                row,
+                column_start,
+                cells,
+                ..
+            } => self.draw_grid_line(row, column_start, cells),
             RedrawEvent::Clear { .. } => self.grid.clear(),
             RedrawEvent::CursorGoto { row, column, .. } => self.cursor.position = (row, column),
-            RedrawEvent::Scroll { top, bottom, left, right, rows, columns, .. } => self.scroll_region(top, bottom, left, right, rows, columns),
+            RedrawEvent::Scroll {
+                top,
+                bottom,
+                left,
+                right,
+                rows,
+                columns,
+                ..
+            } => self.scroll_region(top, bottom, left, right, rows, columns),
+            RedrawEvent::WindowClose { .. } => self.grid.clear(),
```

Comment:
This line is a bit preemptive. I was still learning a bit about how neovide works and figured it might be useful someday to add debugging info here.